### PR TITLE
Performance fixes

### PR DESCRIPTION
**Pull request by sanbornsen**

### CODE REVIEW

**Comment by fabric8cd** [comment]
@sanbornsen snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1984-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@sanbornsen snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1984-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@sanbornsen snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1984-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@sanbornsen snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1984-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@sanbornsen snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1984-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@sanbornsen snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1984-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@sanbornsen snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1984-fabric8-planner.int.rdu2c.fabric8.io

**Comment by nimishamukherjee** [comment]
@sanbornsen will the performance lines be commented out before the merge?

**Comment by sanbornsen** [comment]
@nimishamukherjee no I'd keep them for a while to track.

**Comment by nimishamukherjee** [comment]
@sanbornsen when we do a quick add, the work item type is not displayed on the list view.
missingtype

**Comment by nimishamukherjee** [comment]
editupdate

Applied filter is Sprint 132. Change the work item iteration to Sprint 133. The list still displays the work item even though it does not match the applied filters.

**Comment by sanbornsen** [comment]
@nimishamukherjee Isn't your PR supposed to fix this?

**Comment by sanbornsen** [comment]
@nimishamukherjee first one is fixed now.

**Comment by fabric8cd** [comment]
@sanbornsen snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1984-fabric8-planner.int.rdu2c.fabric8.io

**Comment by vikram-raj** [comment]
screenshot from 2017-06-14 10-52-54

Iterations are loaded but still shows loader and same for areas.

For detail add when I select an iteration then the loader show and never disappear.

File: src/app/components/planner-list/planner-list.component.ts [hunk 1]
```diff
@@ -316,28 +360,13 @@ export class PlannerListComponent implements OnInit, AfterViewInit, DoCheck, OnD
...
-    entryComponent.select();
-    this.selectedWorkItemEntryComponent = entryComponent;
-  }
-
-  onDetail(entryComponent: WorkItemListEntryComponent): void {
-    this.workItemDetail = entryComponent.getWorkItem();
-    this.onSelect(entryComponent);
-    this.showWorkItemDetails = true;
-  }
+  onDetail(entryComponent: WorkItemListEntryComponent): void { }
```
Comment (vikram-raj) [inline src/app/components/planner-list/planner-list.component.ts]: What is the use of this empty function?

File: src/app/components/planner-list/planner-list.component.ts [hunk 2]
```diff
@@ -543,4 +574,8 @@ export class PlannerListComponent implements OnInit, AfterViewInit, DoCheck, OnD
           });
     }
   }
+
+  onSelect() {
+
+  }
```
Comment (vikram-raj) [inline src/app/components/planner-list/planner-list.component.ts]: What is the use of this empty function?

File: src/app/components/work-item-detail/work-item-detail.component.ts [hunk 3]
```diff
@@ -186,30 +189,30 @@ export class WorkItemDetailComponent implements OnInit, AfterViewInit, OnDestroy
...
+        const t2 = performance.now();
+        if (this.panelState === 'out') {
+          this.panelState = 'in';
+          console.log('Performance :: Details page first paint - '  + (t2 - t1) + ' milliseconds.');
+          if (this.headerEditable && typeof(this.title) !== 'undefined') {
+            this.title.nativeElement.focus();
+          }
+        }
+      })
+      .do (workItem => console.log('Work item fethced: ', cloneDeep(workItem)))
```
Comment (vikram-raj) [inline src/app/components/work-item-detail/work-item-detail.component.ts]: Do we need to console log the workItem?

File: src/app/components/work-item-detail/work-item-detail.component.ts [hunk 4]
```diff
@@ -897,6 +828,8 @@ export class WorkItemDetailComponent implements OnInit, AfterViewInit, OnDestroy
   }
 
   iterationUpdated(iterationId: string): void {
+    if (iterationId === '0') return; // Loading item
```
Comment (nimishamukherjee) [inline src/app/components/work-item-detail/work-item-detail.component.ts]: We need to emit the emitEditWI event in the iterationUpdated function as save function is no longer emitting it.

File: src/app/components/work-item-detail/work-item-detail.component.ts [hunk 5]
```diff
@@ -1003,26 +932,46 @@ export class WorkItemDetailComponent implements OnInit, AfterViewInit, OnDestroy
...
+        value: '',
+        selected: false,
+        cssLabelClass: 'spinner spinner-sm spinner-inline'
+      }
+    ];
+    this.getIterations();
   }
 
   areaUpdated(areaId: string) {
-
```
Comment (nimishamukherjee) [inline src/app/components/work-item-detail/work-item-detail.component.ts]: We need to emit the emitEditWI event in the areaUpdated function as save function is no longer emitting it.

File: src/app/components/work-item-detail/work-item-detail.component.ts [hunk 6]
```diff
@@ -850,7 +770,14 @@ export class WorkItemDetailComponent implements OnInit, AfterViewInit, OnDestroy
           }
         }
       });
-      this.save(payload);
```
Comment (nimishamukherjee) [inline src/app/components/work-item-detail/work-item-detail.component.ts]: We need to emit the emitEditWI event in the assignUser function as save function is no longer emitting it.

File: src/app/components/work-item-detail/work-item-detail.component.ts [hunk 7]
```diff
@@ -875,7 +802,11 @@ export class WorkItemDetailComponent implements OnInit, AfterViewInit, OnDestroy
         }
       }
     });
-    this.save(payload);
```
Comment (nimishamukherjee) [inline src/app/components/work-item-detail/work-item-detail.component.ts]: We need to emit the emitEditWI event in the unassignUser function as save function is no longer emitting it.

**Comment by fabric8cd** [comment]
@sanbornsen snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1984-fabric8-planner.int.rdu2c.fabric8.io

File: src/app/components/planner-list/planner-list.component.ts (hunk 1 above)
Comment (sanbornsen) [inline src/app/components/planner-list/planner-list.component.ts]: Nothing basically !! Just kept it there for safety reasons !!

File: src/app/components/planner-list/planner-list.component.ts (hunk 2 above)
Comment (sanbornsen) [inline src/app/components/planner-list/planner-list.component.ts]: Nothing basically !! Just kept it there for safety reasons !!

File: src/app/components/work-item-detail/work-item-detail.component.ts (hunk 3 above)
Comment (sanbornsen) [inline src/app/components/work-item-detail/work-item-detail.component.ts]: Not really, I can remove that.

**Comment by fabric8cd** [comment]
@sanbornsen snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1984-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@sanbornsen snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1984-fabric8-planner.int.rdu2c.fabric8.io

**Pull request merged**